File: src/main/java/de/whitefrog/froggy/exception/RelatedNotPersistedException.java
package de.whitefrog.froggy.exception;

/**
 * Thrown when a relationship should get persisted when a related model is not persisted yet.
 */
public class RelatedNotPersistedException extends PersistException {
  public RelatedNotPersistedException(String message) {
    super(message);
  }
  public RelatedNotPersistedException(String message, Throwable cause) {
    super(message, cause);
  }
}

File: base/src/main/java/de/whitefrog/frogr/exception/RelatedNotPersistedException.kt
package de.whitefrog.frogr.exception

/**
 * Thrown when a relationship should get persisted when a related model is not persisted yet.
 */
class RelatedNotPersistedException : PersistException {
  constructor(message: String) : super(message)
  constructor(message: String, cause: Throwable) : super(message, cause)
}
